dizoo/gym_anytrading/envs/stocks_env.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any
+import numpy as np
+from dizoo.gym_anytrading.envs.trading_env import TradingEnv, Actions, Positions
+from ding.utils import ENV_REGISTRY
+from ding.torch_utils import to_ndarray
+
+
+@ENV_REGISTRY.register('stocks-v0')
+class StocksEnv(TradingEnv):
+
+    def __init__(self, cfg):
+
+        super().__init__(cfg)
+
+        self.trade_fee_bid_percent = 0.01  # unit
+        self.trade_fee_ask_percent = 0.005  # unit
+
+    def _process_data(self, start_idx: int = None) -> Any:
+        '''
+        Overview:
+            used by env.reset(), process the raw data.
+        Arguments:
+            - start_idx (int): the start tick; if None, then randomly select.
+        Returns:
+            - prices: the close.
+            - signal_features: feature map
+        '''
+
+        # ====== build feature map ========
+        all_feature_name = ['Close', 'Open', 'High', 'Low', 'Adj Close', 'Volume']
+        all_feature = {k: self.df.loc[:, k].to_numpy() for k in all_feature_name}
+        # add feature "Diff"
+        prices = self.df.loc[:, 'Close'].to_numpy()
+        diff = np.insert(np.diff(prices), 0, 0)
+        all_feature_name.append('Diff')
+        all_feature['Diff'] = diff
+        # =================================
+
+        # you can select features you want
+        selected_feature_name = ['Close', 'Diff', 'Volume']
+        selected_feature = np.column_stack([all_feature[k] for k in selected_feature_name])
+
+        # validate index
+        if start_idx is None:
+            self.start_idx = np.random.randint(self.window_size, len(self.df) - self._cfg.eps_length - 1)
```

Comment:
the output of `randint(low, high)` falls into [low, high), maybe the right side is too restricted.